leaf-ui/rappid-extensions/LinkInspector.js

```diff
@@ -28,103 +33,172 @@ var LinkInspector = Backbone.View.extend({
+     * 
+     * This function is called on when selected is changed
+     */
+    rerender: function(){
+        var link = this.model.get('link');
+        if (link.get('evolving')) {
+            this.$el.html(_.template($(this.evolvingtemplate).html())(this.model.toJSON()));;
+            $('#link-type-begin').val(link.get("linkType"));
+            this.updateBeginEvolRelations();
+            $('#link-type-end').val(link.get('postType'));
+        }
+    },
+
+    /**
+     * remove the previous view so that we don't have multiple LinkInspector showing up
+     *
+     * This function is called when clearInspector is called
+     */
+    removeView: function(){
+        this.remove();
+    },
+
     /**
      * Switches from Constant Relationship to Evolving Relationship
      * and vice-versa.
      *
      * This function is called on click for #switch-link-type
      */
     switchMode: function() {
-
-        var type = this.cell.attributes.labels[0].attrs.text.text;
-        // array of link values
-        var values = type.split("|");
-
-        this.evolvingRelations = !this.evolvingRelations;
-
-        if (this.evolvingRelations) {
-            this.$el.html(_.template(this.evolvingtemplate)());
-            if (values.length > 1) {
-                this.setSelectValues('#link-type-begin', 'Evolving');
-                $('#link-type-begin').val(values[0].trim()).change();
-                this.updateBeginEvolRelations();
-                $('#link-type-end').val(values[1].trim());
-            } else {
-                $("#link-type-end").prop('disabled', true);
-                $("#link-type-end").css("background-color", "grey");
-                this.setSelectValues('#link-type-begin', 'Evolving');
+        var link = this.model.get('link');
+        var type = link.get('linkType');
+        var postType = link.get('postType');
+        var current = link.get('evolving');
+        link.set('evolving', !current);
+        if(link.get('evolving')){
+            this.$el.html(_.template($(this.evolvingtemplate).html())(this.model.toJSON()));
+            //if the postType is already chosen, meaning not the first time visiting LinkInspector, take the previous values
+            if (postType !== null) { 
+                $('#link-type-begin').val(type); 
+                this.updateBeginEvolRelations(); //makes sure that the available options in the end are consistant with what is allowed if the begin is not changed
+                $('#link-type-end').val(postType);
+            }else{
+                $('#link-type-begin').placeholder= "Begin";
+                $('#link-type-end').placeholder = "End";
             }
-        } else {
-            this.$el.html(_.template(this.template)());
-            $('#constant-links').val(values[0].trim());
-            this.evolvingRelations = false; 
+        }else {
+            link.set('evolving', false);
```

Comment:
I'm confused about why you changed the value of the evolving param, checked it, and then changed it again

Reply:
It is not important, we just took it out.